Component | Rolling Pin Legend

devgru opened the pull request:
This PR introduces a new compact legend component with configurable palette and labels:

![rpl](https://github.com/user-attachments/assets/33e01705-2821-4624-bf01-709ce506381f)

suryahanumandla commented:
@devgru It is always recommended to have dev example for new components. Can you please add it, thanks.

lee00678 commented:
> @devgru It is always recommended to have dev example for new components. Can you please add it, thanks.

We could add a dev example ourselves later. But this one needs documentation. Also, we need an angular component update in this right? cc @rokotyan

Let us know if you need help writing the doc, we have bandwidth this week @devgru

suryahanumandla commented:
> @lee00678 @suryahanumandla
> 
> Thanks for your feedback!
> 
> I'm afraid I don't have any capacity this week as I'm mostly AFK until Monday but I'd be happy to have a look next week!
> 
> I'd start by familiarizing myself with project's internal docs structure, but apart from that I think the effort should be trivial.

@devgru did you get any chance to finish documentation, thanks.

devgru commented:
@lee00678 @suryahanumandla

I think this PR is now ready for review. Thanks for your patience!

Here's what changed since May version:
- rebased the branch to the recent master;
- fixed a couple of minor issues (removed parasitic margins when labels are absent, fixed CSS variable for text color, added actual font-size support);
- added a dev example and a documentation page.

rokotyan commented:
Added section about CSS variables to the docs.

https://github.com/user-attachments/assets/fe899716-ffa8-417e-8ef1-53ad51a3d62e



### Dev app screenshots
<img width="472" alt="SCR-20250708-mjcc" src="https://github.com/user-attachments/assets/9f13d41c-5d82-4dd8-873b-be68f8766d02" />
<img width="436" alt="SCR-20250708-mjdo" src="https://github.com/user-attachments/assets/68392b1c-65cd-47f9-92e9-6b330aa012eb" />

rokotyan commented on packages/react/src/html-components/rolling-pin-legend/index.tsx:
```diff
@@ -0,0 +1,43 @@
+  })
+
+  useImperativeHandle(fRef, () => ({ get component () { return component } }), [])
+  return <div className={props.className} ref={ref} />
```
I think this is out of scope of this PR because this is autogenerated code. It would be nice to add `className` support to other frameworks too, though.

suryahanumandla replied:
thanks @rokotyan, tracking at here https://github.com/f5/unovis/issues/604

rokotyan commented on packages/react/src/html-components/rolling-pin-legend/index.tsx:
```diff
@@ -0,0 +1,43 @@
+
+export type VisRollingPinLegendProps = RollingPinLegendConfigInterface & {
+  ref?: Ref<VisRollingPinLegendRef>;
+  className?: string;
```
Not related to this PR. These descriptions come from autogenerated React props, so we will need to update `packages/react/autogen/component.ts` and add className support to other frameworks; otherwise, the docs are misleading. @suryahanumandla Can you please create a dedicated ticket for this?

suryahanumandla replied:
thanks @rokotyan, tracking at here https://github.com/f5/unovis/issues/604

devgru commented on packages/ts/src/components/rolling-pin-legend/style.ts:
```diff
@@ -0,0 +1,43 @@
+  '--vis-rolling-pin-legend-label-font-size': '12px;',
+  '--vis-rolling-pin-legend-spacing': '4px;',
+  '--vis-rolling-pin-legend-item-width': '8px;',
+  '--vis-dark-rolling-pin-legend-label-color': '#eee;',
```
Fixed, thanks

suryahanumandla reviewed (approved):
LGTM!

suryahanumandla commented:
@devgru can you squash commits? everything is looking great except that, thanks.

devgru commented:
@suryahanumandla sure, will do

devgru commented:
@suryahanumandla sure! could you please merge it? For me merging is still blocked, even with the approve
